packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -784,12 +795,17 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 				}
 				const blobs = {};
 				const localBlobs = new Set<PendingBlob>();
+				// This while is used to protect against blob creation while attaching and getting blobs
 				while (localBlobs.size < this.pendingBlobs.size) {
 					const attachBlobsP: Promise<void>[] = [];
 					for (const [id, entry] of this.pendingBlobs) {
 						if (!localBlobs.has(entry)) {
 							localBlobs.add(entry);
+							// Resolving the blob handle to let hosts continue with their operations (it will resolve
```

Comment:
Does this only happen when "imminent closure" is happening?  Is that worth mentioning as well?